### PR TITLE
pb-2098: Added google file and azure file check to default to KDMP driver.

### PR DESCRIPTION
**Pull request by siva-portworx**
**What type of PR is this?**
bug
**What this PR does / why we need it**:
```
pb-2098: Added google file and azure file check to default to KDMP
    driver.

        - Removed IsDriverWithoutSnapshotSupport and
          isCSISnapshotClassRequired.
        - Added a IsCSIDriverWithoutSnapshotSupport common API in volume pkg
          and adding it in kdmp and csi driver.
        - Also added csiDriverWithOutSnapshotKey configmap field to get user
          input for driver that does not support snapshot.
```
**Does this PR change a user-facing CRD or CLI?**:
no

**Is a release note needed?**:
No

**Does this change need to be cherry-picked to a release branch?**:
no

**Testing:**
Verified the configmap field csiDriverWithOutSnapshotKey by setting it to efs by removed the efs case for testing purpose.
The backup was successful with the kdmp driver.
```
root@ip-70-0-97-59:~# kubectl  describe cm kdmp-config -n kube-system
Name:         kdmp-config
Namespace:    kube-system
Labels:       <none>
Annotations:  <none>

Data
====
KDMP_KOPIAEXECUTOR_REQUEST_CPU:
----
0.1
KDMP_KOPIAEXECUTOR_REQUEST_MEMORY:
----
700Mi
KDMP_MAINTENANCE_JOB_LIMIT:
----
5
KDMP_BACKUP_JOB_LIMIT:
----
5
KDMP_DELETE_JOB_LIMIT:
----
5
KDMP_KOPIAEXECUTOR_IMAGE:
----
portworx/kopiaexecutor:1.0.0
KDMP_KOPIAEXECUTOR_LIMIT_MEMORY:
----
1Gi
CSI_DRIVER_WITHOUT_SNAPSHOT:
----
efs.csi.aws.com
KDMP_KOPIAEXECUTOR_IMAGE_SECRET:
----

KDMP_KOPIAEXECUTOR_LIMIT_CPU:
----
0.2
KDMP_RESTORE_JOB_LIMIT:
----
5
Events:  <none>
root@ip-70-0-97-59:~#
```

### CODE REVIEW

File: drivers/volume/volume.go [hunk 1]
```diff
@@ -689,3 +698,49 @@ func GetPVCFromObjects(objects []runtime.Unstructured, volumeBackupInfo *storkap
...
+		// pure FB csi driver does not support snapshot
+		if driverName == pureCSIProvisioner {
+			if pv.Spec.CSI.VolumeAttributes[pureBackendParam] == pureFileParam {
+				return true
+			}
+		}
+		// vsphere, efs, azure file and google file does not support snapshot.
+		// So defaulting to kdmp by not setting volumesnapshot class.
+		switch driverName {
+		case vSphereCSIProvisioner:
```
Comment (diptiranjanpx) [inline drivers/volume/volume.go:720]: can we make this as a constant list and if driverName is one among those , then will return true.

File: drivers/volume/volume.go [hunk 2]
```diff
@@ -689,3 +698,49 @@ func GetPVCFromObjects(objects []runtime.Unstructured, volumeBackupInfo *storkap
...
+			}
+			if len(kdmpData.Data[csiDriverWithOutSnapshotKey]) != 0 {
+				nameList := strings.Split(kdmpData.Data[csiDriverWithOutSnapshotKey], ",")
+				for _, name := range nameList {
+					if driverName == name {
+						return true
+					}
+				}
+			}
+			return false
```
Comment (diptiranjanpx) [inline drivers/volume/volume.go:742]: this return is not required.

File: drivers/volume/volume.go (hunk 1 above)
Comment (siva-portworx) [reply drivers/volume/volume.go]: Fixed.

File: drivers/volume/volume.go (hunk 2 above)
Comment (siva-portworx) [reply drivers/volume/volume.go]: Fixed

**Review by diptiranjanpx — APPROVED** [review]

**Pull request merged**